test/language/arguments-object/mapped/mapped-arguments-nonwritable-nonconfigurable-descriptors-1.js

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2017 Caio Lima. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+info: Mapped arguments property descriptor change to non-writable and non-configurable
+description: >
+    Mapping just stop working when property is set to non-writable. Change the
+    descriptor using [[DefineOwnProperty]] to {writable: false} and then
+    change property descriptor to {configurable: false}.
+    The descriptor's value is the one set before the property be
+    configured as {writable: false} beacuse mapping is removed.
+flags: [noStrict]
+esid: sec-arguments-exotic-objects-defineownproperty-p-desc
+includes: [propertyHelper.js]
+---*/
+
+function argumentsNonWritableThenNonConfigurableDescriptor(a) {
+  Object.defineProperty(arguments, "0", {writable: false});
+  Object.defineProperty(arguments, "0", {configurable: false});
+  let propertyDescriptor = Object.getOwnPropertyDescriptor(arguments, '0');
+  assert.sameValue(propertyDescriptor.value, 1);
+  verifyEnumerable(arguments, '0');
+  verifyNotWritable(arguments, '0');
+  verifyNotConfigurable(arguments, '0');
```

Comment:
lines 19 to 24 are unnecessary for this test case, just repeating what you should have in another test case (defining to non configurable).

There's a more interesting case here that you can assert the parameter is set to the value if you provide it along with the `writable: false` descriptor, and it becomes unmapped after that. So for example: 

```js
Object.defineProperty(arguments, "0", {writable: false, value: 42});
assert.sameValue(a, 42, 'value is set');

a = 39;
assert.sameValue(a, 39, 'it is still writable');
assert.sameValue(arguments[0], 42, 'and it is unmapped');

arguments[0] = 'foo';
assert.sameValue(a, 39, 'really unmapped');
assert.sameValue(arguments[0], 42, 'non-writable for sure');

...
```